components/common/src/lib.rs

```diff
@@ -52,3 +59,97 @@ lazy_static::lazy_static! {
         }
     };
 }
+
+// TODO (CM): It would be nice to come up with a way to more
+// programmatically manage these flags. It's a bit of a pain to define
+// the flag, and then define the environment variables
+// separately. Nothing statically guarantees that you've specified an
+// variable for a flag.
+
+// TODO (CM): It'd be great to have a built-in way to document them,
+// too.
+
+// TODO (CM): Part of that documentation might be *when* a flag was
+// added. In general, long-lived flags are a code-smell.
+
+// TODO (CM): It may also be useful to break out features by area of
+// concern. We can have any number of bitflags-generated structs.
+
+bitflags::bitflags! {
+    /// All the feature flags that are recogized by Habitat.
+    ///
+    /// In general, feature flags are enabled by setting the corresponding
+    /// environment variable.
+    ///
+    /// Your binary should call `FeatureFlag::from_env` to get a set
+    /// of flags to use.
+    ///
+    /// To add a new feature flag, you will need to add the bit mask
+    /// constant here, as well as a mapping from the feature to the
+    /// environment variable to which it corresponds in the `ENV_VARS`
+    /// map below.
+    pub struct FeatureFlag: u32 {
+        const LIST            = 0b0000_0000_0001;
+        const TEST_EXIT       = 0b0000_0000_0010;
+        const TEST_BOOT_FAIL  = 0b0000_0000_0100;
+        const REDACT_HTTP     = 0b0000_0000_1000;
+        const IGNORE_SIGNALS  = 0b0000_0001_0000;
+        const INSTALL_HOOK    = 0b0000_0010_0000;
+        const OFFLINE_INSTALL = 0b0000_0100_0000;
+        const IGNORE_LOCAL    = 0b0000_1000_0000;
+        const EVENT_STREAM    = 0b0001_0000_0000;
+    }
+}
+
+lazy_static! {
+    static ref ENV_VARS: HashMap<FeatureFlag, &'static str> = {
+        let mapping = vec![(FeatureFlag::LIST, "HAB_FEAT_LIST"),
+                           (FeatureFlag::TEST_EXIT, "HAB_FEAT_TEST_EXIT"),
+                           (FeatureFlag::TEST_BOOT_FAIL, "HAB_FEAT_BOOT_FAIL"),
+                           (FeatureFlag::REDACT_HTTP, "HAB_FEAT_REDACT_HTTP"),
+                           (FeatureFlag::IGNORE_SIGNALS, "HAB_FEAT_IGNORE_SIGNALS"),
+                           (FeatureFlag::INSTALL_HOOK, "HAB_FEAT_INSTALL_HOOK"),
+                           (FeatureFlag::OFFLINE_INSTALL, "HAB_FEAT_OFFLINE_INSTALL"),
+                           (FeatureFlag::IGNORE_LOCAL, "HAB_FEAT_IGNORE_LOCAL"),
+                           (FeatureFlag::EVENT_STREAM, "HAB_FEAT_EVENT_STREAM")];
+        HashMap::from_iter(mapping)
+    };
+}
+
+impl FeatureFlag {
+    /// If the environment variable for a flag is set to _anything_ but
+    /// the empty string, it is activated.
+    pub fn from_env(ui: &mut UI) -> Self {
+        let mut flags = FeatureFlag::empty();
+
+        for (feature, env_var) in ENV_VARS.iter() {
+            if henv::var(env_var).is_ok() {
```

Comment:
Can we use use `std::env::var_os` and `is_some` instead?

Reply:
Yup, I just realized I forgot about that... adding it now.